src/modules/compliance/src/lib/procedures/EnsureFilePermissions.cpp

```diff
@@ -342,4 +332,121 @@ REMEDIATE_FN(EnsureFilePermissions, "filename:Path to the file:M", "owner:Requir
     OsConfigLogDebug(log, "File '%s' remediation succeeded", filename.c_str());
     return Status::Compliant;
 }
+
+Result<Status> EnsureFilePermissionsCollectionHelper(std::map<std::string, std::string> args, IndicatorsTree& indicators, ContextInterface& context, bool isRemediation)
+{
+    auto log = context.GetLogHandle();
+    auto it = args.find("directory");
+    if (it == args.end())
+    {
+        OsConfigLogError(log, "No directory provided");
+        return Error("No directory provided", EINVAL);
+    }
+    auto directory = std::move(it->second);
+    it = args.find("ext");
+    if (it == args.end())
+    {
+        OsConfigLogError(log, "No file pattern provided");
+        return Error("No file pattern provided", EINVAL);
+    }
+    auto ext = std::move(it->second);
+
+    auto directorycharp = std::unique_ptr<char, void (*)(void*)>(strdup(directory.c_str()), free);
+    if (nullptr == directorycharp)
+    {
+        OsConfigLogError(log, "Failed to allocate memory for directory path");
+        return Error("Failed to allocate memory for directory path", ENOMEM);
+    }
+    char* const paths[] = {directorycharp.get(), nullptr};
```

Comment:
Could this be replaced by:
```suggestion
    char* const paths[] = {&directory[0], nullptr};
```
?

Reply:
I was way too worried about the constness. Fixed.